backend/controllers/event.controller.ts

```diff
@@ -1,13 +1,277 @@
+import { AppPermissionName, Event, EventStatus } from "@prisma/client";
+import {
+  CursorPaginationDatetimeParams,
+  CursorPaginationDatetimeSchema,
+  EventCreateSchema,
+  EventUpdateSchema,
+  IdParamSchema,
+} from "../../shared/zodSchemas";
 import { NextFunction, Request, Response } from "express";
-import { IdParamSchema } from "../../shared/zodSchemas";
 import prisma from "../prisma/client"; // import the singleton prisma instance
 import { AppError, AppErrorName } from "../utils/AppError";
+import { checkUserPermission } from "../utils/checkUserPermission";
+
+const userId = 3; // Placeholder for testing
 
 // test Event
 export const eventTest = async (req: Request, res: Response) => {
   res.status(200).json({ message: "Event endpoint works" });
 };
 
+// Create new Event
+// Takes multipart form data.
+// The user is required to have CREATE_EVENTS permission for the organization
+export const createVerifiedEvent = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    // Get userId
+    // const userId = req.user.id; // get userId from the request -> should set in auth middleware
```

Comment:
Would you be able to actually set this up inside of backend/middleware/verifyAuth in the PR and update this to actually use it?

Reply:
req.body.userId